### PR TITLE
[Transaction] Fix transaction sequenceId generate error.

### PR DESCRIPTION
**Pull request by congbobo184**
## Motivation
now, we recover transaction sequenceId from lastConfirmEntry or managedLedger properties. But update managedLedger properties is not a Synchronize op, so it will recover error sequenceId. 

### Modifications
use ManagedLedgerInterceptor will fix this problem.
1. use interceptor.onUpdateManagedLedgerInfo, when ledger roll over, it will update current sequenceId to managedLedger properties.
2. use interceptor.onManagedLedgerPropertiesInitialize, first we will recover sequenceId from managedLedger properties.
3. use interceptor.onManagedLedgerLastLedgerInitialize, when tc recover and has a effective lastConfirmEntry, we will change the sequenceID after interceptor.onManagedLedgerPropertiesInitialize
4. If a new TC, it can't recover from interceptor.onManagedLedgerPropertiesInitialize and interceptor.onManagedLedgerLastLedgerInitialize, it will use the initial sequenceId -1
### Verifying this change
Add the tests for it

Does this pull request potentially affect one of the following parts:
If yes was chosen, please highlight the changes

Dependencies (does it add or upgrade a dependency): (no)
The public API: (no)
The schema: (no)
The default values of configurations: (no)
The wire protocol: (no)
The rest endpoints: (no)
The admin cli options: (no)
Anything that affects deployment: (no)

### CODE REVIEW

**Review by eolivelli — COMMENTED** [review]
is this change compatible with existing Pulsar clusters ?

**Comment by congbobo184** [comment]
> is this change compatible with existing Pulsar clusters ?

It will not.

**Comment by congbobo184** [comment]
> is this change compatible with existing Pulsar clusters ?

It will be compatible

**Review by eolivelli — APPROVED** [review]
LGTM

File: pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/MLTransactionMetadataStoreProvider.java [hunk 1]
```diff
@@ -42,10 +42,14 @@
...
+        MLTransactionLogInterceptor mlTransactionLogInterceptor = new MLTransactionLogInterceptor();
+        managedLedgerConfig.setManagedLedgerInterceptor(new MLTransactionLogInterceptor());
         MLTransactionLogImpl txnLog = new MLTransactionLogImpl(transactionCoordinatorId,
                 managedLedgerFactory, managedLedgerConfig);
 
+        // MLTransactionLogInterceptor will init sequenceId and update the sequenceId to managedLedger properties.
         return txnLog.initialize().thenApply(__ ->
-                new MLTransactionMetadataStore(transactionCoordinatorId, txnLog, timeoutTracker, recoverTracker));
+                new MLTransactionMetadataStore(transactionCoordinatorId, txnLog, timeoutTracker,
+                        recoverTracker, mlTransactionLogInterceptor.getSequenceId()));
```
Comment (codelipenghui) [inline pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/MLTransactionMetadataStoreProvider.java:53]: It's better to introduce modify sequence ID method for `mlTransactionLogInterceptor`, share AtomicLong across multiple instances is not good for maintainance

**Pull request merged**